akka-bootstrapper/src/main/scala/filodb/akkabootstrapper/ClusterSeedDiscovery.scala

```diff
@@ -24,40 +24,44 @@ abstract class ClusterSeedDiscovery(val cluster: Cluster,
   }
 
   @throws(classOf[DiscoveryTimeoutException])
+  //scalastyle:off null
   protected def discoverPeersForNewCluster: Seq[Address]
 
   protected def discoverExistingCluster: Seq[Address] = {
 
     val seedsEndpoint = settings.seedsBaseUrl + settings.seedsPath
+    var response: HttpResponse[String] = null
+    var retriesRemaining = settings.seedsHttpRetries
+    do {
+      try {
+        logger.info(s"Trying to fetch seeds from $seedsEndpoint ... $retriesRemaining retries remaining.")
+        response = Http(seedsEndpoint).timeout(2000, 2000).asString
+        logger.info(s"Seeds endpoint returned a ${response.code}. Response body was ${response.body}")
+      } catch {
+        case NonFatal(e) =>
+          logger.error(s"Seeds endpoint $seedsEndpoint call threw an exception", e)
+      }
+      Thread.sleep(settings.seedsHttpSleepBetweenRetries.toMillis)
```

Comment:
Isn't it better to add an exponential backoff ?

Reply:
This is during bootstrap, initial formation of cluster for the very first time. We should form the soon and not back-off exponentially.